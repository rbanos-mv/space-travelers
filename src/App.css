* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  min-height: 100vh;
}

nav {
  padding: 1rem 2rem;
}

.navbar-content {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid #999;
  padding: 1rem;
}

.logo-container {
  display: flex;
  align-items: center;
}

.logo {
  width: 64px;
  height: 64px;
  margin-right: 1rem;
}

.app-title {
  display: inline-block;
}

.nav-link {
  text-decoration: none;
  color: #3c85fe;
  padding: 1rem;
}

.activated {
  text-decoration: underline;
}

.pages-content {
  padding: 1rem 3rem;
}

ul {
  list-style-type: none;
}

.column {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.row {
  display: flex;
  flex-direction: row;
  align-content: center;
}

.rocket-card {
  justify-content: flex-start;
  margin-bottom: 1.5rem;
}

.rocket-info {
  align-items: flex-start;
  justify-content: flex-start;
  margin-left: 1.5rem;
}

.rocket-name {
  padding-bottom: 0.5rem;
}

.rocket-description {
  font-size: 16px;
  padding-bottom: 0.5rem;
}

.cancel-btn,
.reserve-btn {
  border-radius: 5px;
  height: 40px;
  width: 200px;
}

.cancel-btn {
  background-color: white;
  border: 1px solid lightslategrey;
  color: lightslategrey;
}

.reserve-btn {
  background-color: #0290ff;
  border: 1px solid #0290ff;
  color: white;
}
