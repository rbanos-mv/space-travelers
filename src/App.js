import { HashRouter as Router, Route, Routes } from 'react-router-dom';
import './App.css';
import Navbar from './components/Navbar';
import Missions from './pages/Missions';
import MyProfile from './pages/MyProfile';
import Rockets from './pages/Rockets';

function App() {
  return (
    <Router>
      <div className="App">
        <Navbar />
        <Routes>
          <Route path="/" element={<Rockets />} />
          <Route path="missions" element={<Missions />} />
          <Route path="my-profile" element={<MyProfile />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
