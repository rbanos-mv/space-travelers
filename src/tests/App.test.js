import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../App';
import setupStore from './setupStore';

describe('Test the App.test components', () => {
  it('renders the App correctly', () => {
    const store = setupStore(false);
    const app = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });

  it('shows the Rockets page when clicking the navbar link', () => {
    const store = setupStore(false);
    const { container } = render(
      <Provider store={store}>
        <App url="/" />
      </Provider>,
    );
    fireEvent.click(screen.getByText('Missions'));
    fireEvent.click(screen.getByText('Rockets'));
    expect(container.getElementsByClassName('rockets-page')).toHaveLength(1);
  });

  it('shows the Missions page when clicking the navbar link', () => {
    const store = setupStore(false);
    const { container } = render(
      <Provider store={store}>
        <App url="/" />
      </Provider>,
    );
    fireEvent.click(screen.getByText('Missions'));
    expect(container.getElementsByClassName('missions-page')).toHaveLength(1);
  });

  it('shows the My Profile page when clicking the navbar link', () => {
    const store = setupStore(false);
    const { container } = render(
      <Provider store={store}>
        <App url="/" />
      </Provider>,
    );
    fireEvent.click(screen.getByText('My Profile'));
    expect(container.getElementsByClassName('profile')).toHaveLength(2);
  });
});
