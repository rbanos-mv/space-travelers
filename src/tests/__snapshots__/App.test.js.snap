// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App.test components renders the App correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav>
          <div
            class="navbar-content"
          >
            <div
              class="logo-container"
            >
              <img
                alt=""
                class="logo"
                src="logo.png"
              />
              <h1
                class="app-title"
              >
                Space Travelers' Hub
              </h1>
            </div>
            <div
              class="nav-links"
            >
              <a
                aria-current="page"
                class="nav-link activated"
                href="#/"
              >
                Rockets
              </a>
              <a
                class="nav-link"
                href="#/missions"
              >
                Missions
              </a>
              <a
                class="nav-link"
                href="#/my-profile"
              >
                My Profile
              </a>
            </div>
          </div>
        </nav>
        <div
          class="rockets-page"
        >
          <ul>
            <li
              class="row rocket-card"
            >
              <img
                alt="{rocketName}"
                height="auto"
                src="https://imgur.com/DaCfMsj.jpg"
                width="250"
              />
              <div
                class="column rocket-info"
              >
                <h3
                  class="rocket-name"
                >
                  Falcon 1
                </h3>
                <p
                  class="rocket-description"
                >
                  The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
                </p>
                <button
                  class="reserve-btn"
                  id="1"
                  type="button"
                >
                  Reserve Rocket
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav>
        <div
          class="navbar-content"
        >
          <div
            class="logo-container"
          >
            <img
              alt=""
              class="logo"
              src="logo.png"
            />
            <h1
              class="app-title"
            >
              Space Travelers' Hub
            </h1>
          </div>
          <div
            class="nav-links"
          >
            <a
              aria-current="page"
              class="nav-link activated"
              href="#/"
            >
              Rockets
            </a>
            <a
              class="nav-link"
              href="#/missions"
            >
              Missions
            </a>
            <a
              class="nav-link"
              href="#/my-profile"
            >
              My Profile
            </a>
          </div>
        </div>
      </nav>
      <div
        class="rockets-page"
      >
        <ul>
          <li
            class="row rocket-card"
          >
            <img
              alt="{rocketName}"
              height="auto"
              src="https://imgur.com/DaCfMsj.jpg"
              width="250"
            />
            <div
              class="column rocket-info"
            >
              <h3
                class="rocket-name"
              >
                Falcon 1
              </h3>
              <p
                class="rocket-description"
              >
                The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
              </p>
              <button
                class="reserve-btn"
                id="1"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
